scheduler/pkg/kafka/gateway/worker.go

```diff
@@ -312,7 +312,7 @@ func (iw *InferWorker) restRequest(ctx context.Context, job *InferWork, maybeCon
 	}
 	req, err := http.NewRequestWithContext(ctx, http.MethodPost, restUrl.String(), bytes.NewBuffer(data))
 	if err != nil {
-		return err
+		return iw.produce(ctx, job, iw.topicNamer.GetModelErrorTopic(), []byte(err.Error()), true, nil)
```

Comment:
`iw.produce(...)` returns an error only when `iw.consumer.Produce(...)` returns an error.  The latter calls the Kafka api ([here](https://github.com/SeldonIO/seldon-core/blob/294b5f823fc5784f49157639a3e6e97b2617a460/scheduler/pkg/kafka/gateway/infer.go#L160)). That `kc.producer.Produce(msg, deliveryChan)` is an asynchronous call - it merely enqueues the message in internal kafka client buffers, then returns.

We pass a deliveryChan to it, and we would receive any actual errors on that channel. However, at the moment we're [just ending the trace span when we get an event on the channel](https://github.com/SeldonIO/seldon-core/blob/294b5f823fc5784f49157639a3e6e97b2617a460/scheduler/pkg/kafka/gateway/worker.go#L290). 

I think we should also check there if the returned Kafka event on that channel shows an error or not, and log it if it does.

Something along the lines of:
```
 e := <-deliveryChan
 span.End()
 m := e.(*kafka.Message)

 if m.TopicPartition.Error != nil {
     // log error
 }

 close(deliveryChan)
 ```